### DISCUSSION TITLE
Using Spring View Component with JTE Spring Boot: Making Them Work Together

### DISCUSSION THREAD

**Discussion by robertellam** [tschuehly/spring-view-component#10] [Q&A]
Can I run regular Spring Boot JTE templates together with Spring View Components? I'm in the process of integrating Spring View Components into my existing Spring Boot Java application that uses JTE, but I'm facing some challenges in getting everything to function properly. If I decide to leverage Spring View Components, does that mean I have to convert all my existing JTE templates into components as well? Alternatively, is it feasible to invoke Spring View Components from within my current JTE templates?

**Comment by tschuehly** [comment]
Hello @robertellam,

I think the problem is that the CodeResolver in DEV mode only looks in the src/main/resources https://github.com/tschuehly/spring-view-component/issues/8#issuecomment-1608449423

In Thymeleaf it works out of the box, so I need to add the CodeResolver that @DevAlves1993 created in Issue 8.

I'm currently working on a build time processor that needs to compile jte inline and maybe your problem will be solved with that anyway: https://github.com/tschuehly/spring-view-component/blob/v0.6.2-SNAPSHOT/core/src/main/kotlin/de/tschuehly/spring/viewcomponent/core/processor/ViewComponentProcessor.kt

**Comment by jnellis** [comment]
Responding pretty late here so maybe this is a non-issue now. I've gotten this to work as long as all your non-ViewComponent .jte are *directly* in src/main/java. 

Another issue I've found is that gradle jte plugin block requires specifying the `sourceDirectory` otherwise as soon as you make `spring.view-component.local-development=false` in application.properties, everything breaks (regardless if you have non-ViewComponent .jtes)

* in build.gradle (groovy)
```
dependencies {
    implementation 'org.springframework.boot:spring-boot-starter-web'
    implementation 'gg.jte:jte:3.1.12'
    implementation 'gg.jte:jte-spring-boot-starter-3:3.1.12'
    implementation 'io.github.wimdeblauwe:htmx-spring-boot:4.0.1'

    implementation("de.tschuehly:spring-view-component-jte:0.8.3")
    developmentOnly("org.springframework.boot:spring-boot-devtools") // !!
    annotationProcessor("org.springframework.boot:spring-boot-configuration-processor")

    testImplementation 'org.springframework.boot:spring-boot-starter-test'
    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
}

jte {
    generate()
    binaryStaticContent = true

    sourceDirectory = file("src/main/java").toPath()
}
```
* in application.properties
```
gg.jte.development-mode=false 
gg.jte.templateLocation=src/main/java

spring.application.name=jteDemo

# following causes 404 no static resource error
# when this is set to false and sourceDirectory 
# in gradle jte block is not set
spring.view-component.local-development=false

spring.view-component.viewComponentRoot=src/main/java
spring.devtools.livereload.enabled=false
```

```java
@Controller
public class DemoController {

  private final SimpleViewComponent simpleViewComponent;

  public DemoController(SimpleViewComponent simpleViewComponent) {this.simpleViewComponent = simpleViewComponent;}

  @GetMapping("/")
  public String view(Model model, HttpServletResponse response){
    model.addAttribute("greeting", "Hello Whirled" );
    return "demo";
  }

  @GetMapping("/vc")
  public ViewContext vc(){
    return simpleViewComponent.render();
  } 
}
```
